lib/rule-doc-notices.ts

```diff
@@ -197,6 +199,43 @@ const RULE_NOTICES: {
     ruleName,
     urlRuleDoc,
   }) => {
+    // use object type `DeprecatedInfo`
+    if (typeof deprecatedInfo === 'object') {
+      const replacementRuleList = (deprecatedInfo.replacedBy ?? [])
+        .map(({ rule }) =>
+          rule && rule.name
+            ? rule.url
+              ? `[\`${rule.name}\`](${rule.url})`
+              : `\`${rule.name}\``
+            : undefined,
+        )
+        .filter((rule): rule is string => typeof rule === 'string');
+
+      return `${EMOJI_DEPRECATED} This rule is deprecated${
+        deprecatedInfo.deprecatedSince
+          ? ` since v${deprecatedInfo.deprecatedSince}.`
+          : '.'
+      }${
+        replacementRuleList.length > 0
+          ? ` It was replaced by ${String(replacementRuleList)}.`
+          : ''
+      }${
+        // use DeprecatedInfo#url to inform about the reasons
+        deprecatedInfo.url
+          ? `${EOL}${EOL}Read more at [${new URL(deprecatedInfo.url).hostname}](${deprecatedInfo.url})`
+          : ''
+      }`;
+    }
+
+    // warn and use fallback
+    console.warn(
+      [
+        'The two top-level properties `deprecated` and `replacedBy` are deprecated since eslint 9.21.0.',
+        'Please consider using the new object type `DeprecatedInfo`.',
+        'https://eslint.org/docs/latest/extend/rule-deprecation#-deprecatedinfo-type',
+      ].join('\n'),
+    );
+
```

Comment:
Actually, while it's a nice idea to warn about this, I'd rather use a new lint rule in eslint-plugin-eslint-plugin to do this:

- https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/517 

eslint-doc-generator is not really intended to warn about deprecations or older styles. That's the job of eslint-plugin-eslint-plugin.